dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -685,18 +691,25 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
       memset(nvjpeg_destinations_.data(), 0, nvjpeg_destinations_.size() * sizeof(nvjpegImage_t));
 
       int j = 0;
+      TensorVector<CPUBackend> tv(samples_hw_batched_.size());
+
       for (auto *sample : samples_hw_batched_) {
         assert(!sample->roi);
         int i = sample->sample_idx;
         const auto &in = ws.Input<CPUBackend>(0, i);
         const auto &out_shape = output_shape_.tensor_shape(i);
 
-        in_data_[j] = in.data<uint8_t>();
+        tv[j].ShareData(const_cast<Tensor<CPUBackend>*>(&in));
         in_lengths_[j] = in.size();
         nvjpeg_destinations_[j].channel[0] = output.mutable_tensor<uint8_t>(i);
         nvjpeg_destinations_[j].pitch[0] = out_shape[1] * out_shape[2];
         j++;
       }
+
+      hw_decoder_images_staging_.Copy(tv, 0);
```

Comment:
This is a potential hazard:
```suggestion
      hw_decoder_images_staging_.Copy(tv, hw_decode_stream_);
```
...and it's quite likely that this copy should go _after_ `CUDA_CALL(cudaEventSynchronize(hw_decode_event_));` to make sure that the pinned buffer has been consumed before we overwrite it.
Alternatively, you can use double buffering for the staging area.

Reply:
Fixed